surrogates/common_utils/fits.py

```diff
@@ -0,0 +1,139 @@
+    # multiply surrogate amplitude with overall normalization factor
+    h_approx = np.conj(np.array(h_approx))*norm
+    
+    return h_approx
+
+
+#----------------------------------------------------------------------------------------------------
+def all_modes_surrogate(modes, X_input, fit_data_dict_1, fit_data_dict_2, \
+                        B_dict_1, B_dict_2, lmax, fit_func, decomposition_funcs, norm):
+
+    """ Takes the fit data (either from splines or GPR), matrix B and computes the 
+        interpolated waveform for all modes 
+        
+    Inputs
+    ======
+    
+        modes : list of modes to evaluate
+        
+        X_input :  array of surrogate parameterization e.g. [log(q), spin1, spin2]
+
+        fit_data_dict_1, fit_data_dict_2 : dictionary of fit data obtained for two datapieces from 
+                                           the h5 file.
+                                           Keys are the modes.
+                                           Structure may depend on ether the data comes from spline 
+                                           fits or GPR fits. However, they should always be packed 
+                                           in fit_data_dict_1 and fit_data_dict_2. 
+                                           Make sure to modify your data loading script to achieve
+                                           this if necessary.
+
+        B_dict_1, B_dict_2 : dictionary of the basis matrices obtained from h5 file.
+                             Modes used as keys.
+
+        fit_func : form of fitting function. options : 'spline_1d' or 'GPR_fits'
+
+        decomposition_funcs : form of data decomposition function to combine datapieces for 22 and
+                              higher modes respectively. e.g. Amp/Phase to full or real/imag to full
+                              etc. These functions are available at common_utils.utils.py
+
+        norm : overall normalization factor to be multiplied to final waveform. This depends on the 
+              way the surrogate have been constructed. Mostly norm=1/q or norm=1. 
+    
+    Outputs
+    =======
+    
+        t_surrogate : time array
+        h_surrogate : dictiornary of modes  
+        
+    """
+    
+    # dictionary to save waveform
+    h_approx_dict={}
+    # evaluate all the modes     
+    for mode in modes:
+        (l,m) = mode
+        # load modes only upto l=lmax
+        if l<=lmax:
+            # get the fit data for specific mode for both the datapieces
+            fit_data_1 = fit_data_dict_1[mode]
+            fit_data_2 = fit_data_dict_2[mode]
+            
+            # read the decompositon function for the modes; special treatment for the
+            # 22 mode and higher order modes
+            if mode==(2,2):
+                decomposition_func = decomposition_funcs[0]
+            else:
+                decomposition_func = decomposition_funcs[1]
+    
+            # evaluate surrogate modes
```

Comment:
Is my understanding correct that for models using the co-orbital frame, the modes are still in the co-oorbital frame at this point? And the rotation back to the inertial frame is in evaluate_surrogate? If so, adding that code comment here would be helpful.

Reply:
Yes. At this stage, for models using co-orbital frame, modes are still in the co-orbital frame.

Even evaluate _surrogate or all_modes_surrogate do not transform the mode back to inertial frame. It is done after that using utils.obtain_processed_output.
Here: 
https://github.com/tousifislam/BHPTNRSurrogate/blob/main/surrogates/model_utils/eval_surrogates.py#L93

Reply:
I have added the comment in the doc string as well as in the code comment